AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/fragments/movieDetails/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.androidhomeapplication.fragments.movieDetails
+
+import android.util.Log
+import androidx.lifecycle.*
+import com.android.academy.fundamentals.homework.data.MovieRepository
+import com.example.androidhomeapplication.DataResult
+import com.example.androidhomeapplication.models.Movie
+import kotlinx.coroutines.launch
+
+class MovieDetailsViewModel(
+    private val movieRepository: MovieRepository,
+    private val movieId: Long?
+) : ViewModel() {
+
+    private val mutableMovie = MutableLiveData<DataResult<Movie>>(DataResult.Default())
+    val movie: LiveData<DataResult<Movie>> get() = mutableMovie
+
+    init {
+        if (movieId != null) {
+            getMovieDetails(movieId)
+        }
+    }
+
+    fun getMovieDetails(movieId: Long) {
```

Comment:
not good practice to use class value as parameter inside class
u can access it without param